src/main/java/graphql/validation/rules/UniqueArgumentNamesRule.java

```diff
@@ -0,0 +1,64 @@
+package graphql.validation.rules;
+
+import graphql.language.Argument;
+import graphql.language.Directive;
+import graphql.language.Field;
+import graphql.language.Node;
+import graphql.validation.AbstractRule;
+import graphql.validation.ValidationContext;
+import graphql.validation.ValidationErrorCollector;
+import graphql.validation.ValidationErrorType;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Unique argument names
+ *
+ * A GraphQL field or directive is only valid if all supplied arguments are uniquely named.
+ */
+public class UniqueArgumentNamesRule extends AbstractRule {
+    public UniqueArgumentNamesRule(ValidationContext validationContext, ValidationErrorCollector validationErrorCollector) {
+        super(validationContext, validationErrorCollector);
+    }
+
+    @Override
+    public void checkField(Field field) {
+        if(field.getArguments()==null||field.getArguments().size()<=1){
+            return;
+        }
+
+        Set<String> arguments=new HashSet<>(field.getArguments().size());
```

Comment:
sizing collections is almost never needed.  but easier to read code is.

I wont say not to this PR because of this but its not needed

Reply:
sizing collections is in order to avoid the cost for `resize() in HashMap/HashSet`, but I haven took loadFactor into consideration. In fact, I want to init Map:
```
  int capacity = (int) (field.getArguments().size() / 0.75 + 1);
```
But default capacity of HashMap is enough in most case, i will remove this and make it clear.

Thanks for your code review.